boxen/instance/qemulaunchcmd.go

```diff
@@ -136,6 +136,17 @@ func (i *Qemu) launchCmdCPU() []string {
 			}...)
 	}
 
+	// when only cores are specified, the `-smp %s` gets appended
+	if c.Cores != 0 && c.Threads == 0 && c.Sockets == 0 {
+		if len(cpuCmd) == 0 {
+			cpuCmd = append(cpuCmd, []string{"-cpu", "max"}...)
+		}
+
+		cpuCmd = append(
+			cpuCmd,
+			[]string{"-smp", fmt.Sprint(c.Cores)}...)
+	}
+
```

Comment:
the idea here was that not always you want to specify cores and threads. I saw many times that VMs were defined just with `-smp 4` to denote that 4 vcpu are required.

Reply:
sorry missed replying to this one before. could we do something like this?

```go
	if c.Cores != 0 {
		if len(cpuCmd) == 0 {
			cpuCmd = append(cpuCmd, []string{"-cpu", "max"}...)
		}

		if c.Threads != 0 && c.Sockets != 0 {
			cpuCmd = append(
				cpuCmd,
				[]string{
					"-smp",
					fmt.Sprintf("cores=%d,threads=%d,sockets=%d", c.Cores, c.Threads, c.Sockets),
				}...)
		} else if c.Threads == 0 && c.Sockets == 0 {
			cpuCmd = append(
				cpuCmd,
				[]string{"-smp", fmt.Sprint(c.Cores)}...)
		}

```

and just to refresh my memory -- basically this is just going to say "hey if you gimme cores and nothing else we'll give you smp blah" whereas if you give me cores/sockets/threads we'll do the whole shebang like before ya?

Reply:
good call, avoids duplication

done in [52a562d](https://github.com/carlmontanari/boxen/pull/33/commits/52a562dee0d66d78b50eb71ab64efffeff9c047c)


yes, the idea is to simplify multi core provisioning when it is enough just to specify I want X cores, and don't know how many sockets/threads I need